lib/phoenix_live_view/helpers.ex

```diff
@@ -269,6 +271,22 @@ defmodule Phoenix.LiveView.Helpers do
     end
   end
 
+  defmacro component(_func, _assigns, _do_block) do
+    raise ArgumentError, """
+      component/3 expected an anonymous function with 1-arity
+
+      Please call component with a 1-arity function, for example:
+
+          <%= component &func/1 %>
+
+          def func(assigns) do
+            ~L\"""
+            Hello
+            \"""
+          end
+    """
+  end
```

Comment:
We should not validate this at the AST level. Instead, there is already a clause that checks if it is a function, let's make sure we are also checking the arity there.

Reply:
So this will work to avoid BadArityError [`98158bf` (#1429)](https://github.com/phoenixframework/phoenix_live_view/pull/1429/commits/98158bf66d2c3f67b8d2c9c5d6d1cea6b7207bb4) but calling a 0 arity fn will still raise a `(CompileError) [...] expected "assigns" to expand to an existing variable or be part of a match`, that's why I tried to check on the AST level, before expanding.

Reply:
FWIW, I cannot reproduce the assigns error.

Reply:
Sorry I didn't provide more info, let me give an example:

It's not an error on LV, but a way to help someone that write something like this by mistake:

```elixir
  defmodule RenderOnly do
    use Phoenix.LiveComponent

    def render(assigns) do
      ~L"""
      <%= component &hello/0 %>
      """
    end

    defp hello do
      ~L"""
      Hello
      """
    end
  end
```

or

```elixir
  defmodule RenderOnly do
    use Phoenix.LiveComponent

    def render(assigns) do
      ~L"""
      <%= component &hello/2, name: "WORLD" %>
      """
    end

    defp hello(assigns, foo) do
      ~L"""
      Hello <%= @name %>
      """
    end
  end
```

Just trying to provide a better experience :)